livre/eq/settings/CameraSettings.cpp

```diff
@@ -25,76 +25,99 @@
 #include <livre/core/maths/maths.h>
 #include <co/co.h>
 
+#include <algorithm>
+
 namespace livre
 {
 
 CameraSettings::CameraSettings()
-    : modelRotation_( Matrix4f::IDENTITY )
-    , cameraPosition_( 0.f, 0.f, 1.f )
 {
+    Matrix4f initial = Matrix4f::IDENTITY;
+    std::copy( &initial.array[0], &initial.array[0] + 16, getMatrix( ));
 }
 
-void CameraSettings::serialize( co::DataOStream& os, const uint64_t dirtyBits )
+void CameraSettings::spinModel( const float x, const float y )
 {
-    co::Serializable::serialize( os, dirtyBits );
-    os << modelRotation_ << cameraPosition_;
-}
+    if( x == 0.f && y == 0.f )
+        return;
 
-void CameraSettings::deserialize( co::DataIStream& is, const uint64_t dirtyBits )
-{
-    co::Serializable::deserialize( is, dirtyBits );
-    is >> modelRotation_ >> cameraPosition_;
-}
+    float matrixValues[16];
+    std::copy( getMatrix(), getMatrix() + 16, matrixValues );
 
-void CameraSettings::spinModel( const float x, const float y, const float z )
-{
-    if( x == 0.f && y == 0.f && z == 0.f )
-        return;
+    Matrix4f modelview( &matrixValues[0], &matrixValues[0] + 16 );
 
-    modelRotation_.pre_rotate_x( x );
-    modelRotation_.pre_rotate_y( y );
-    modelRotation_.pre_rotate_z( z );
-    setDirty( DIRTY_ALL );
+    modelview[0][3] = 0.0f;
+    modelview[1][3] = 0.0f;
+    modelview[2][3] = 0.0f;
+
+    modelview.pre_rotate_x( x );
+    modelview.pre_rotate_y( y );
+
+    modelview[0][3] = matrixValues[12];
+    modelview[1][3] = matrixValues[13];
+    modelview[2][3] = matrixValues[14];
+
+    std::copy( &modelview.array[0], &modelview.array[0] + 16, getMatrix( ));
```

Comment:
Same problem as the comment before. Need to use std::copy.
"setMatrix( modelview.array );" gives:
"error: no matching function for call to 'livre::CameraSettings::setMatrix(float [16])'"